src/idl_gen_js_ts.cpp

```diff
@@ -1859,66 +1871,68 @@ class JsTsGenerator : public BaseGenerator {
       GenerateFinisher(struct_def, code_ptr, code, object_name, true);
 
       // Generate a convenient CreateX function
-      if (lang_.language == IDLOptions::kJs) {
-        std::string paramDoc =
-            GenTypeAnnotation(kParam, "flatbuffers.Builder", "builder");
-        for (auto it = struct_def.fields.vec.begin();
-             it != struct_def.fields.vec.end(); ++it) {
-          const auto &field = **it;
-          if (field.deprecated) continue;
+      if (CanCreateFactoryMethod(struct_def)) {
+        if (lang_.language == IDLOptions::kJs) {
+          std::string paramDoc =
```

Comment:
can you make sure to use `snake_case` and `auto` where possible

Reply:
I only changed the indentation level here :)

Reply:
Ok, fair enough..